internal/request/graphql/schema/types/types.go

```diff
@@ -121,31 +123,24 @@ func PolicyDirective() *gql.Directive {
 	})
 }
 
-func IndexDirective(orderingEnum *gql.Enum) *gql.Directive {
-	return gql.NewDirective(gql.DirectiveConfig{
-		Name:        IndexDirectiveLabel,
-		Description: "@index is a directive that can be used to create an index on a type.",
-		Args: gql.FieldConfigArgument{
-			IndexDirectivePropName: &gql.ArgumentConfig{
+func IndexFieldInputObject(orderingEnum *gql.Enum) *gql.InputObject {
+	return gql.NewInputObject(gql.InputObjectConfig{
+		Name: "IndexField",
+		Fields: gql.InputObjectConfigFieldMap{
```

Comment:
todo: Please add a Description, and note the implicit/explicit behaviour on it when attached to a field.

These descriptions are important and get rendered in the docs section and intellisense of GQL clients, and I guess that ORMs will also use them to document bound types.

Reply:
done